src/docfx/config/Config.cs

```diff
@@ -146,97 +146,111 @@ public static Config Load(string docsetPath, CommandLineOptions options, bool ex
         /// Load the config if it exists under <paramref name="docsetPath"/>
         /// </summary>
         /// <returns>Whether config exists under <paramref name="docsetPath"/></returns>
-        public static bool LoadIfExists(string docsetPath, CommandLineOptions options, out Config config, bool extend = true, RestoreMap restoreMap = null)
+        public static bool LoadIfExists(List<Error> errors, string docsetPath, CommandLineOptions options, out Config config, bool extend = true, RestoreMap restoreMap = null)
```

Comment:
How about `(errors, config) LoadIfExists(...)`? when it does not exist, return `(errors, null)`

Reply:
Thought about that as well, but we do use the bool return value to check if config file exists to control diff flows. So seems better to keep the signature and pass in errors list

---
In reply to: [211462329](https://github.com/dotnet/docfx/pull/3243#discussion_r211462329) [](ancestors = 211462329)

Reply:
or if can be `bool TryLoad(out var (errors, config))`, just don't let it populate an errors list in an API boundary like here, it is easy to introduce mutli threading problems like below.

---
In reply to: [211463455](https://github.com/dotnet/docfx/pull/3243#discussion_r211463455) [](ancestors = 211463455,211462329)

Reply:
ok, sounds better

---
In reply to: [211463951](https://github.com/dotnet/docfx/pull/3243#discussion_r211463951) [](ancestors = 211463951,211463455,211462329)